scrapyelasticsearch/scrapyelasticsearch.py

```diff
@@ -125,11 +130,23 @@ def index_item(self, item):
         elif index_suffix_key:
             index_name += "-" + index_suffix_key
 
-        index_action = {
-            '_index': index_name,
-            '_type': self.settings['ELASTICSEARCH_TYPE'],
-            '_source': dict(item)
-        }
+        item_type = self.settings['ELASTICSEARCH_TYPE']
+        item_dict = dict(item)
+
+        if self.settings['ELASTICSEARCH_MERGE'] == True:
+            index_action = {
+                '_op_type': 'update',
+                '_index': index_name,
+                '_type': item_type,
+                'doc': item_dict,
+                'upsert': item_dict
```

Comment:
Couldn't you use `"doc_as_upsert": true` instead of sending the `item_dict` twice here? (Just trying to understand your patch, beginner at Elasticsearch over here.)

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#doc_as_upsert